File: backend/OlivePlatform/OlivePlatform.Application/Features/Harvests/Responses/HarvestDetailsResponse.cs


using OlivePlatform.Domain.Entities;
using OlivePlatform.Domain.Enums;

namespace OlivePlatform.Application.Features.Harvests.Responses
{
    public class HarvestDetailsResponse
    {
        public int Id { get; set; }
        public string Reference { get; set; } = string.Empty;
        public string PlotReference { get; set; } = string.Empty;
        public DateOnly HarvestDate { get; set; }
        public decimal QuantityKg { get; set; }
        public int PlannedTrees { get; set; }
        public required int VarietyId { get; set; }
        public int HarvestedTrees { get; set; }
        public string? Notes { get; set; }
        public ProductionStatus Status { get; set; }
        public DateTime? StartTime { get; set; }
        public DateTime? EndTime { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }

        public List<HarvestStockForListResponse> Stocks { get; set; } = [];
    }
}

File: frontend/olive-platform/src/features/harvests/data/responses/HarvestDetailsResponse.ts
import type { ProductionStatus } from "../../../production/domain/entities/ProductionStatus";
import type { OliveVarieties } from "../../../shared/entities/OliveVarieties";
import type { HarvestStockResponse } from "./HarvestStockResponse";

export interface HarvestDetailsResponse {
  id: number;
  reference: string;
  plotId: number;
  variety: OliveVarieties;
  harvestedTrees: number;
  plannedTrees: number;
  harvestDate: string;
  quantityKg: number | null;
  notes: string | null;
  createdAt: string;
  updatedAt: string | null;
  status: ProductionStatus;
  startTime: string | null;
  endTime: string | null;
  stocks?: HarvestStockResponse[];
}